### PR TITLE
refactor(local): use libuv to create symlinks

### PR DESCRIPTION
**Pull request by nanotee**
This PR replaces the use of external programs to create symlinks with the built-in `vim.loop.fs_symlink()` and adds a test for local plugins

EDIT: I've verified that this works on Windows 8.1 while running neovim-qt as administrator. I'm actually unsure whether developer mode on Windows 10 allows all programs to create symlinks or just `mklink`.

EDIT2: Apparently `libuv` calls the `CreateSymbolicLink` API with the `SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE` flag so this should work in theory, but I don't have a copy of Windows 10 to test this with

### CODE REVIEW

**Review by wbthomason — APPROVED** [review]
This looks great, thanks. I am in general all for moving toward `libuv`-based functionality rather than using external programs.

I also unfortunately do not have a Windows 10 environment to test this in. It's probably OK to merge, but I'm going to see if we can solicit Windows testers in the `packer` Gitter first. If nobody has responded within a day, we'll merge and hope for the best.

I don't think it needs to be part of this PR, but I should augment the test workflow to also run on Windows and macOS environments, to prevent this uncertainty in the future.

**Comment by nanotee** [comment]
Thanks for the feedback. This probably isn't super urgent since it doesn't fix bugs or add new features. I'm familiarizing myself with the codebase in the process

**Comment by akinsho** [comment]
@nanotee I just noticed that the new test files are all prefixed with `packer_` I had just created a `plugin_utils_spec.lua` then noticed there was already a packer prefixed version.

What do you think about dropping the prefixes? I don't really think they are necessary since tbh depending on how complex things get we can actually just match packers own structure using directories (rather than prefixes) so tests are always in a predictable place?

But I don't think that's necessary to start but I do think it will be easier to find test files if they just match the actual filename + `_spec.lua`?

**Comment by nanotee** [comment]
I wasn't sure if there was a specific naming convention so I stuck to `packer_*_spec.lua`, dropping the prefix definitely makes sense to me :+1:

**Comment by wbthomason** [comment]
We haven't gotten any Win10 testers, but (given that @nanotee tested on Win8, at least), I think this is OK to merge. @nanotee, any reason you'd like to hold off for now?

**Comment by nanotee** [comment]
I'd like to rebase this PR on `master` once #340 gets merged since there were a few changes in the tests

**Comment by wbthomason** [comment]
@nanotee Please see the discussion on #340: I discovered a bug after merging and quickly reverted, so we'll need to fix the bug, re-merge, and then cleanup the git history.

**Comment by wbthomason** [comment]
@nanotee the new PR to rebase onto is #374

**Comment by nanotee** [comment]
Okay, rebase done. `packer_plugin_utils_spec.lua` should probably get merged into `plugin_utils_spec.lua` at some point but I thought it was out of scope for this PR.

**Comment by wbthomason** [comment]
Sorry for taking so long to merge this!

**Pull request merged**
